multimap/src/test/java/org/infinispan/multimap/impl/DistributedMultimapCacheTest.java

```diff
@@ -0,0 +1,382 @@
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertFalse;
+import static org.testng.AssertJUnit.assertNotNull;
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.infinispan.Cache;
+import org.infinispan.commons.dataconversion.Encoder;
+import org.infinispan.commons.dataconversion.EncodingUtils;
+import org.infinispan.commons.dataconversion.Wrapper;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.container.DataContainer;
+import org.infinispan.container.entries.CacheEntry;
+import org.infinispan.container.entries.ImmortalCacheEntry;
+import org.infinispan.container.entries.InternalCacheEntry;
+import org.infinispan.distribution.BaseDistFunctionalTest;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.multimap.api.EmbeddedMultimapCacheManagerFactory;
+import org.infinispan.multimap.api.MultimapCache;
+import org.infinispan.multimap.api.MultimapCacheManager;
+import org.infinispan.remoting.transport.Address;
+import org.testng.annotations.Test;
+
+@Test(groups = "functional", testName = "distribution.DistributedMultimapCacheTest")
+public class DistributedMultimapCacheTest extends BaseDistFunctionalTest<String, Collection<User>> {
+
+   protected Map<Address, MultimapCache<String, User>> multimapCacheCluster = new HashMap<>();
+
+   protected boolean fromOwner;
+
+   public DistributedMultimapCacheTest fromOwner(boolean fromOwner) {
+      this.fromOwner = fromOwner;
+      return this;
+   }
+
+   @Override
+   protected String[] parameterNames() {
+      return concat(super.parameterNames(), "fromOwner");
+   }
+
+   @Override
+   protected Object[] parameterValues() {
+      return concat(super.parameterValues(), fromOwner ? Boolean.TRUE : Boolean.FALSE);
+   }
+
+   @Override
+   public Object[] factory() {
+      List testsToRun = new ArrayList<>();
+      testsToRun.add(new DistributedMultimapCacheTest().fromOwner(false).cacheMode(CacheMode.DIST_SYNC).transactional(false));
+      testsToRun.add(new DistributedMultimapCacheTest().fromOwner(true).cacheMode(CacheMode.DIST_SYNC).transactional(false));
+      return testsToRun.toArray();
+   }
+
+   @Override
+   protected void createCacheManagers() throws Throwable {
+      super.createCacheManagers();
+
+      for (EmbeddedCacheManager cacheManager : cacheManagers) {
+         MultimapCacheManager multimapCacheManager = EmbeddedMultimapCacheManagerFactory.from(cacheManager);
+         multimapCacheCluster.put(cacheManager.getAddress(), multimapCacheManager.get(cacheName));
+      }
+   }
+
+   @Override
+   protected void initAndTest() {
+      assertMultimapCacheSize(multimapCacheCluster, 0);
+
+      putValuesOnMultimapCache(multimapCacheCluster, NAMES_KEY, OIHANA);
```

Comment:
Why are you inserting the value on all nodes? If this is clustered, you should do it only once.